### PR TITLE
SPI/LCD check code

### PR DESCRIPTION
**Pull request by alees24**
Introduce lcd_check which drives the SPI controller and LCD at maximum speed to exercise/test the reliability of the core/FIFO logic.
Support transposed rendering and output in the SPI LCD model.
~~This depends on PR #198~~

### CODE REVIEW

File: sw/cheri/checks/lowrisc_logo_native.h
Comment (marnovandermaas) [inline sw/cheri/checks/lowrisc_logo_native.h:1]: How did you generate this?

**Review by marnovandermaas — COMMENTED** [review]
Looks reasonable to me. Especially in the checks folder it is nice to have something like this that keeps running. We should fix CI before merging though.

File: sw/cheri/checks/lowrisc_logo_native.h
Comment (alees24) [reply sw/cheri/checks/lowrisc_logo_native.h:1]: Not exactly; it's just converted into the native LCD format from the file `sw/legacy/demo/lcd_st7735/lowrisc_logo.h` I've added a comment to that effect.

**Comment by AlexJones0** [comment]
@alees24 Here is the fix to Nix to get this working in CI:

```patch
diff --git a/nix/software.nix b/nix/software.nix
index 9224241e..bbaa45ab 100644
--- a/nix/software.nix
+++ b/nix/software.nix
@@ -19,13 +19,14 @@ in {
   sonata-system-software = pkgs.stdenv.mkDerivation rec {
     name = "sonata-system-software";
     src = fileset.toSource {
-      root = ../sw;
+      root = ../.;
       fileset = fileset.unions [
         ../sw/cheri
         ../sw/common
+        ../vendor/display_drivers/st7735
       ];
     };
-    sourceRoot = "${src.name}/cheri";
+    sourceRoot = "${src.name}/sw/cheri";
     nativeBuildInputs = cheriotPkgs ++ (with pkgs; [cmake]);
     cmakeFlags = [
       "-DFETCHCONTENT_SOURCE_DIR_CHERIOT_RTOS=${cheriotRtosSource}"
(nix:sonata-system-devshell-env) [15:56:14] [alexj] [~/lr/sonata-syst
```

**Comment by alees24** [comment]
> @alees24 Here is the fix to Nix to get this working in CI:
> ...

Thanks for that help.

File: sw/cheri/checks/lowrisc_logo_native.h
Comment (marnovandermaas) [reply sw/cheri/checks/lowrisc_logo_native.h:1]: Thanks!

**Review by marnovandermaas — APPROVED** [review]
Tested this after a rebase on main so I think this is ready to merge.

File: dv/dpi/spidpi/spi_lcd.hh [hunk 1]
```diff
@@ -45,6 +46,8 @@ private:
 
   bool pixWriting;  // Pixel writing mode?
 
+  uint8_t  madCtl;  // Memory Data Access Control.
```
Comment (HU90m) [inline dv/dpi/spidpi/spi_lcd.hh:49]: Should this be `mdaCtrl`?

**Pull request merged**